src/org/etools/j1939_84/bus/RP1210Bus.java

```diff
@@ -165,7 +128,7 @@ public RP1210Bus(RP1210Library rp1210Library,
             sendCommand(CMD_ECHO_TRANSMITTED_MESSAGES, ECHO_ON);
             sendCommand(CMD_SET_ALL_FILTERS_STATES_TO_PASS);
 
-            this.exec.submit(this::poll);
+            this.rp1210Executor.submit(this::poll);
```

Comment:
I realize this is executing on a thread pool.  Should it have it’s own dedicated executor?

Reply:
I don't understand your question.  All calls to rp1210 need to be from the same thread. rp1210Executor is a single thread executor.  This way sendMessage(), sendCommand() and readMessage() will never overlap.  The only rp1210 calls outside of rp1210Executor are the configuration calls in the constructor.

Reply:
ok.  I didn't realize they all needed to be in the same thread.  In that case, should there be two threads in the pool?